c/extras/aggregator.cc

```diff
@@ -641,6 +667,25 @@ void Aggregator::group_nd(const dtptr& dt, dtptr& dt_members) {
 }
 
 
+void Aggregator::fill_coprimes(size_t n, std::vector<size_t>& coprimes) {
+  coprimes.clear();
+  for (size_t i = 1; i <= n; ++i) {
+    if (gcd(i, n) == 1) coprimes.push_back(i);
```

Comment:
Eratosthenes Sieve is a much better approach to finding coprimes:
- create a bitmask of length `n`, filled with 0s;
- starting with `i = 2`:
  - if `mask[i]` is 1 then continue;
  - if `n` is divisible by `i`, then set all elements of the bitmask that are multiples of `i` to 1;
- continue until `i*i < n`.
In the end, all elements of bitmask that are 0s are coprime to `n`.

Reply:
Yes, sure, current approach can be optimized, I was more concerned about the final `members_count` distribution for the moment. For the aggregation purpose we can even stick to the list of prime numbers that are smaller than the typical `nd_max_bins`, i.e. ~1000, and then just use these prime numbers as the "seed".

Reply:
Actually, it seems that we can just use any prime number that is larger than `nd_max_bins`.

Reply:
Yes, sorry. I probably kept in mind the approach of `[n / 2; n]` for the range of prime numbers.

Reply:
Ok, I've changed it to Eratosthenes sieve, stopping at `i == n / 2`. Could probably optimized even further, but still good for our purpose.